ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/StartDebugParameters.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api.debug;
+
+import java.io.File;
+import java.util.List;
+import org.netbeans.api.extexecution.ExecutionDescriptor;
+import org.openide.util.Lookup;
+
+/**
+ * Parameters passed to {@link NIDebugger#start(org.netbeans.modules.nativeimage.api.debug.StartDebugParameters, java.util.function.Consumer)}.
+ *
+ * @since 0.5
+ */
+public final class StartDebugParameters {
```

Comment:
Final class forms a good [open space](http://wiki.apidesign.org/wiki/ClientAPI) and it can easily grow over time - you'll be able to add new parameters to it.